### PR TITLE
[issue-659]Handle for function "substring"

### PR DESCRIPTION
**Pull request by Jay357089**
issue #659 

the error is because that for aggregation Expressison, field in subString is not resolved, and cause that query order is not increased and make that  sum(age) show the value of "productdate" , which can not be cast to double, that is null and sum(gamepointid) shows the value of sum(age).

### CODE REVIEW

**Comment by Jay357089** [comment]
@ravipesala @kumarvishal09  @sraghunandan pls check.

**Comment by Jay357089** [comment]
@QiangCai pls review

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/aggquery/FunctionQueryTestCase.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      Seq(Row("2015-07", 14.0, 60010.0),
+        Row("2015-08",123.0, 45120.0),
+        Row("2015-09", 120.0, 30118.0),
+        Row("2015-10", 185.0, 30183.0))
+    )
+  }
+
+  test("group by with subString(String)") {
+    checkAnswer(
+      sql( """SELECT substring(phonetype,1,6), sum(ID), sum(salary) FROM carbonTable
```
Comment (QiangCai) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/aggquery/FunctionQueryTestCase.scala]: please try to use substring(phonetype,1,6) as col1

**Comment by Jay357089** [comment]
@QiangCai  checked and add test case. pls reivew.

**Comment by Jay357089** [comment]
@ravipesala   pls review if any free time.

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/aggquery/FunctionQueryTestCase.scala (hunk 1 above)
Comment (QiangCai) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/aggquery/FunctionQueryTestCase.scala]: please try：concat(substring(phonetype,1,7), 'plus')

**Comment by ravipesala** [comment]
@Jay357089 This issue happens not only for substring, it could be any function like trim or length etc. Please verify in the master, this issue might have already fixed.

**Comment by ravipesala** [comment]
This issue is fixed by PR https://github.com/HuaweiBigData/carbondata/pull/645

**Comment by Jay357089** [comment]
OK. close it.

**Pull request closed without merging**